### PR TITLE
Add, change and fix some mappings

### PR DESCRIPTION
**Pull request by ozelot379**
- `category`: Podcast Category
- `hdVideo`: iTunes Video Quality (2=Full HD, 1=HD, 0=SD)
- `keywords`: Podcast Keywords
- `movement`: Movement
- `movementIndex`: Movement Index/Total, e.g. `{no: 1, of: 2}`
- `movementTotal`: The total number of movements
- `podcastId`: Podcast Identifier
- `showMovement`: Show Movement
- `stik`: iTunes Media Type (1=Normal, 2=Audiobook, 6=Music Video, 9=Movie, 10=TV Show, 11=Booklet, 14=Ringtone https://github.com/sergiomb2/libmp4v2/wiki/iTunesMetadata#user-content-media-type-stik
- Make `subtitle` as `multiple` to match return type `string[]` (May a breaking change)
- Add missing `description` type
- Make `compilation` and `podcast` `boolean` type (`string` before) (May a breaking change)
- Ignore generated `test/samples/zeroes.flac`
- `metadata.common.composersort` has been changed to a singleton (May a breaking change)

ID3:
- Set ID3v22 as `CaseInsensitiveTagMap`, ID3v24 is already (May a breaking change)
- Add mapping `MVNM` to `movement`
- Add mapping `MVIN` (1 field) to `movementIndex`
- Add mapping `PCS` and `PCST` to `podcast` (Already exists for MP4)
- Add mapping `TCAT` to `category`
- Add mapping `TCP` to `compilation` (Already exists for ID3>2.2 and for MP4)
- Add mapping `TDES` to `description`
- Add mapping `TDR` and `TDRL` to `date` (Already exists for MP4)
- Add mapping `TGID` to `podcastId`
- Add mapping `TKWD` to `keywords`
- Add mapping `TS2` to `albumartistsort` (Already exists for ID3>2.2 and for MP4)
- Add mapping `TSA` to `albumsort` (Already exists for ID3>2.2 and for MP4)
- Add mapping `TSC` to `composersort` (Already exists for ID3>2.2 and for MP4)
- Add mapping `TSP` to `artistsort` (Already exists for ID3>2.2 and for MP4)
- Add mapping `TST` to `titlesort` (Already exists for ID3>2.2 and for MP4)
- Add mapping `WFD` and `WFED` to `podcasturl` (Already exists for MP4)
- Change mapping `TOT` from `work` to `originalalbum` (May a breaking change)
- The follow fields seems not to exists like in MP4:
  - `hdVideo`
  - `showMovement`
  - `stik`
  - `work`

MP4:
- Set as `CaseInsensitiveTagMap` (May a breaking change)
- Add mapping `©com` to `comment`
- Add mapping `©cpy` to `copyright`
- Add mapping `©mvn` to `movement`
- Add mapping `©mvi`/`©mvc` (2 fields) to `movementIndex` and `movementTotal`
- Add mapping `©wrk` to `work`
- Add mapping `catg` to `category`
- Add mapping `egid` to `podcastId`
- Add mapping `hdvd` to `hdVideo`
- Add mapping `keyw` to `keywords`
- Add mapping `shwm` to `showMovement`
- Add mapping `stik` to `stik`
- Change mapping `desc` from `description` to `subtitle` for differentiate with `ldesc` (May a breaking change)

### CODE REVIEW

File: lib/common/GenericTagTypes.ts [hunk 1]
```diff
@@ -106,7 +106,16 @@ export type GenericTagId =
...
+  | 'description'
+  | 'category'
+  | 'hdvideo'
+  | 'keywords'
+  | 'movement'
+  | 'movementnumber'
+  | 'movementnumbertotal'
+  | 'podcastid'
+  | 'showmovement'
+  | 'stik';
```
Comment (Borewit) [inline lib/common/GenericTagTypes.ts:118]: Please also update corresponding the API documentation: [doc-gen/common.json](https://github.com/Borewit/music-metadata/blob/master/doc-gen/common.json).

I know _common_ properties have an inconsistent naming style (one of the reasons is backward compatibility), but please use apply camel case for new ones.

File: lib/mp4/MP4Parser.ts [hunk 2]
```diff
@@ -298,6 +301,15 @@ export class MP4Parser extends BasicParser {
   }
 
   private addTag(id: string, value: any) {
+    if (id === "©mvi") {
```
Comment (ozelot379) [inline lib/mp4/MP4Parser.ts:304]: @Borewit Some idea to do this better?

In MP4 there are 2 number fields, in ID3 only one (`"x/y"`)

**Comment by Borewit** [comment]
Thanks you so much for the documentation, that looks very good @ozelot379 .
I need to find some time review the PR in detail.

There are some small lint errors, you could possibly fix them in the mean time (`yarn run lint`).

File: doc-gen/common.json [hunk 3]
```diff
@@ -298,5 +298,29 @@
...
+  "hdVideo": {
+    "description": "iTunes Video Quality (2=Full HD, 1=HD, 0=SD)"
+  },
+  "keywords": {
+    "description": "Podcast Keywords"
+  },
+  "movement": {
+    "description": "Movement"
+  },
+  "movementNumber": {
```
Comment (Borewit) [inline doc-gen/common.json:314]:
```suggestion
  "movementIndex": {
```

File: doc-gen/common.json [hunk 4]
```diff
@@ -298,5 +298,29 @@
...
+    "description": "Movement Number/Total, e.g. `{no: 1, of: 2}`"
```
Comment (Borewit) [inline doc-gen/common.json:315]:
```suggestion
    "description": "Movement Index/Total, e.g. `{no: 1, of: 2}`"
```

File: lib/id3v2/ID3v24TagMapper.ts [hunk 5]
```diff
@@ -136,7 +136,17 @@ const id3v24TagMap: INativeTagMap = {
   "TXXX:REPLAYGAIN_ALBUM_GAIN": "replaygain_album_gain",
   "TXXX:MP3GAIN_MINMAX": "replaygain_track_minmax",
   "TXXX:MP3GAIN_ALBUM_MINMAX": "replaygain_album_minmax",
-  "TXXX:MP3GAIN_UNDO": "replaygain_undo"
+  "TXXX:MP3GAIN_UNDO": "replaygain_undo",
+
+  MVNM: "movement",
+  MVIN: "movementNumber",
```
Comment (Borewit) [inline lib/id3v2/ID3v24TagMapper.ts:142]:
```suggestion
  MVIN: "movementIndex",
```

File: lib/common/GenericTagTypes.ts [hunk 6]
```diff
@@ -231,7 +239,16 @@ export const commonTags: ITagInfoMap = {
...
   replaygain_undo: {multiple: false},
 
-  description:  {multiple: true}
+  description:  {multiple: true},
+
+  category: {multiple: true},
+  hdVideo: {multiple: false},
+  keywords: {multiple: true},
+  movement: {multiple: false},
+  movementNumber: {multiple: false},
```
Comment (Borewit) [inline lib/common/GenericTagTypes.ts:248]:
```suggestion
  movementIndex: {multiple: false},
```

File: lib/common/MetadataCollector.ts [hunk 7]
```diff
@@ -67,7 +67,8 @@ export class MetadataCollector implements INativeMetadataCollector {
 
   public readonly common: ICommonTagsResult = {
     track: {no: null, of: null},
-    disk: {no: null, of: null}
+    disk: {no: null, of: null},
+    movementNumber: {no: null, of: null}
```
Comment (Borewit) [inline lib/common/MetadataCollector.ts:71]:
```suggestion
    movement: {}
```

Comment (Borewit) [reply lib/common/MetadataCollector.ts:71]: The nulls are inherited from the abandoned project [musicmetadata](https://github.com/leetreveil/musicmetadata), `undefined` is sufficient.

File: lib/mp4/MP4TagMapper.ts [hunk 8]
```diff
@@ -7,6 +7,7 @@ import {CommonTagMapper} from '../common/GenericTagMapper';
 const mp4TagMap: INativeTagMap = {
   '©nam': 'title',
   '©ART': 'artist',
+  '©art': 'artist',
```
Comment (Borewit) [inline lib/mp4/MP4TagMapper.ts:10]: Should the tag mapping be case insensitive?

File: lib/mp4/MP4Parser.ts (hunk 2 above)
Comment (ozelot379) [reply lib/mp4/MP4Parser.ts]: Fixed

File: lib/mp4/MP4TagMapper.ts (hunk 8 above)
Comment (ozelot379) [reply lib/mp4/MP4TagMapper.ts]: You're right, it's a better idea, I made it now case insensitive

File: test/test-pr-544.ts [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+            t.strictEqual(metadata.common.hdVideo, 2, 'metadata.common.hdVideo');
+            t.strictEqual(metadata.common.stik, 9, 'metadata.common.stik');
+        });
+    });
+
+    function testEqualsSame(metadata: mm.IAudioMetadata): void {
+        t.strictEqual(metadata.common.albumartistsort, 'Album Artist Sort', 'metadata.common.albumartistsort');
+        t.strictEqual(metadata.common.albumsort, 'Album Sort', 'metadata.common.albumsort');
+        t.strictEqual(metadata.common.artistsort, 'Artist Sort', 'metadata.common.artistsort');
+        t.deepEqual(metadata.common.composersort, ['Composer Sort'], 'metadata.common.composersort'); // TODO: Why composer sort field is an array but other sort fields not?
```
Comment (ozelot379) [inline test/test-pr-544.ts:68]: @Borewit 
Why composer sort field is an array but other sort fields not?

Comment (Borewit) [reply test/test-pr-544.ts:68]: Hmm, I have no idea. Good point.

**Comment by ozelot379** [comment]
@Borewit 
I added an unit test.

But I had to change the follow too to make it work:

> - Make `subtitle` as `multiple` to match return type `string[]`
> - Add missing `description` type
> - Make `compilation` and `podcast` `boolean` type (`string` before)

I think it's a small breaking change, but at least the other unit tests are not broken

**Comment by Borewit** [comment]
@ozelot379. I think there is room for improvement in the unit tests. I have no problem with a reference to the original issue, but the unit test should describe which feature it is testing.

But this is all absolutely awesome work @ozelot379.

**Comment by ozelot379** [comment]
@Borewit 
If I run `build` or `test` command, it wil generate some changes in `doc/common_metadata.md` and generate a new file `test/samples/zeroes.flac`

Is this correct and should I commit these files too?

File: test/test-pr-544.ts (hunk 9 above)
Comment (ozelot379) [reply test/test-pr-544.ts:68]: @Borewit 
Should I change to make `composersort` not `multiple` (=additional breaking change) or keep it as is?

Both works in my case

**Comment by Borewit** [comment]
> @Borewit
> If I run `build` or `test` command, it generates some changes in `doc/common_metadata.md` and generates a new file `test/samples/zeroes.flac`
> 
> Is this correct and should I commit these files too?

`doc/common_metadata.md`, yes please, check it in. It is used to generate documentation.
`test/samples/zeroes.flac`, should be ignored.

File: test/test-pr-544.ts (hunk 9 above)
Comment (Borewit) [reply test/test-pr-544.ts]: Please correct it, and add it to the list of changes: _metadata.common.composersort has been changed to a singleton_

**Comment by Borewit** [comment]
Part of [v7.0.0](https://github.com/Borewit/music-metadata/releases/tag/v7.0.0)

**Comment by ozelot379** [comment]
:rocket:

**Pull request merged**